src/WebJobs.Script/CloudBlockBlobHelperService.cs

```diff
@@ -4,19 +4,22 @@
 using System;
 using System.Threading.Tasks;
 using Microsoft.Azure.Storage.Blob;
-using Microsoft.Extensions.Logging;
 
-namespace Microsoft.Azure.WebJobs.Script.WebHost.Management.LinuxSpecialization
+namespace Microsoft.Azure.WebJobs.Script
 {
-    public class RunFromPackageCloudBlockBlobService
+    public class CloudBlockBlobHelperService
     {
-        public virtual async Task<bool> BlobExists(string url, string environmentVariableName, ILogger logger)
+        public virtual async Task<bool> BlobExists(string url)
         {
-            return await BlobExistsAsync(url, environmentVariableName, logger);
+            return await BlobExistsAsync(url);
```

Comment:
change to `return BlobExistsAsync`

Reply:
BlobExists is async so without await this results in a compiler error

Reply:
remove async from the method signature :)

Reply:
But it is still an async method? We have:
`await Utility.InvokeWithRetriesAsync(async () => (...)`

Reply:
change to :

```cs
public virtual Task<bool> BlobExists(string url)
        {
            if (string.IsNullOrEmpty(url))
            {
                return Task.FromResult(false);
            }
            return BlobExistsAsync(url);
        }
```

Its good to avoid `async .. await` when result is not needed and is simple used to return the result.

Reply:
Didn't know that, thanks for letting me know! Updated.